### DISCUSSION TITLE
Project ENV Improvements

### DISCUSSION THREAD

**Discussion by JarrodMFlesch** [vercel/community#1183] [Help]
We have a project that has 2 domains, a staging and a prod domain. The staging domain is setup as the preview deployment, and deploys when our stage branch updates. The prod domain is attached to the production deployment.

When setting ENV variables, it is difficult to manage them on the settings page. It becomes quite cluttered with just a few env variables because you must duplicate them to adjust what environments they apply to. 

I am just wondering if this has been brought up before? I searched the discussions but did not see a topic similar.

Below is an image that illustrates how 3 variables turns into 6 rows, and they are always in order of last saved, meaning they can get out of order making it even harder to work with.

![CleanShot 2022-12-21 at 15 57 28](https://user-images.githubusercontent.com/30633324/209004587-39e5369c-ff2b-40a9-a1be-85395fb550c3.png)

**Comment by amyegan** [comment]
Thanks for bringing this up @JarrodMFlesch. Do you have any ideas about what you would like to see instead of the current UI?

**Reply by JarrodMFlesch** [reply]
I feel like it could look just like it does now, but the edit screen would open up and allow you to set values specific to each environment. 

So instead of the edit view looking like this:

![CleanShot 2022-12-21 at 16 51 13](https://user-images.githubusercontent.com/30633324/209009560-7dd41a60-237c-4824-85c9-26471f98e1d4.png)


It would be more table-like:

| ENV checkboxes                 | Value        |
|----------------------|--------------|
| [x] Production     | prod-super-secret |
| [x] Development      | dev-super-secret |
| [x] Preview     | prev-super-secret |


Now obviously that comes with more UI/UX discovery, i.e. unset 1, remove all, etc.

But I think it would make this much easier for people jumping into this kind of environment.
